File: SecondSem/CN/LAB/Java/ConnectionLessFullDuplex_4/UDPClients.java
import java.net.*;

public class UDPClients {
    public static void main(String[] args) {
        try {
            DatagramSocket clientSocket = new DatagramSocket();

            InetAddress IPAddress = InetAddress.getByName("localhost");

            byte[] sendData;
            byte[] receiveData = new byte[1024];

            while (true) {
                String sentence = "Hello from client!";
                sendData = sentence.getBytes();

                DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, IPAddress, 9876);
                clientSocket.send(sendPacket);

                DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
                clientSocket.receive(receivePacket);

                String modifiedSentence = new String(receivePacket.getData(), 0, receivePacket.getLength());
                System.out.println("Server response: " + modifiedSentence);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: SecondSem/CN/LAB/Python/ConnectionLessFullDuplex_4/UDPclients.py
import socket


def main():
    try:
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        server_address = ("localhost", 9876)

        while True:
            sentence = "Hello from client!"
            send_data = sentence.encode()

            client_socket.sendto(send_data, server_address)

            receive_data, _ = client_socket.recvfrom(1024)
            modified_sentence = receive_data.decode()
            print("Server response: " + modified_sentence)
    except Exception as e:
        print(e)


if __name__ == "__main__":
    main()
